deployer/commands/transform/cost_table.py

```diff
@@ -0,0 +1,163 @@
+import re
+from pathlib import Path
+
+import pandas as pd
+import typer
+
+from deployer.cli_app import transform_app
+from deployer.utils.rendering import print_colour
+
+# Creates a new typer application, which is then nested as a sub-command named
+# "cost-table" under the `transform` sub-command of the deployer.
+cost_table_app = typer.Typer(pretty_exceptions_show_locals=False)
+transform_app.add_typer(
+    cost_table_app,
+    name="cost-table",
+    help="Transform the cost table from a cloud provider when completing billing cycles",
+)
+
+
+@cost_table_app.command()
+def aws(
+    input_path: Path = typer.Argument(
+        ..., help="The file path to the cost table downloaded from the AWS UI"
+    ),
+    output_path: Path = typer.Option(
+        None,
+        help="(Optional) The path to write the output CSV to. If None, one will be constructed.",
+    ),
+):
+    """
+    Ingests a CSV cost table generated via the AWS UI and performs a transformation.
+
+    We assume the input CSV has a column for each linked account and rows for:
+    - the linked account ID and name
+    - a monthly total for each month selected
+    - a linked account total across the month range
+
+    We aim to have an output CSV file with columns:
+    - linked account name
+    - start-month
+    - ...
+    - end-month
+    - linked account total
+    """
+    # Read the CSV file into a pandas dataframe. Skip the first row and this
+    # contains numerical project IDs - the project names begin on the second row.
+    # Rename the columns so that they are all lower case and any whitespace is
+    # replaced with underscores. Then strip '_($)' to ensure we just have the
+    # project names.
+    df = pd.read_csv(
+        input_path,
+        skiprows=1,
+    ).rename(columns=lambda col: col.strip().lower().replace(" ", "_").strip("_($)"))
+
+    # Ensure values of the linked_account_name column are lower case and any
+    # whitespace is replaced with underscores.
```

Comment:
```suggestion
    # Ensure values of the linked_account_name column are lower case and any
    # whitespace is replaced with underscores.
    # Note that because AWS outputs the linked account names as rows, this
    # column name is misleading - we are not affecting linked account names here.
```